### ISSUE TITLE
Make first runtime argument parameter named

### ISSUE THREAD

**Issue by ilyapuchka** [AliSoftware/Dip#18]
With introduction of runtime arguments (#8) the API became a bit confusing cause of mixing tags and arguments values. It would be better to use something like this: 

```swift
container.register(tag: "tag") { (arg1: Arg1) in ... as Protocol }
container.resolve(tag: "tag", withArguments: arg1) as Protocol
```

To release this feature faster and without bumping major version (what we are going to do to release circular dependencies) we can leave old version and later (or can we do it right away?) deprecate it.

**Comment by AliSoftware** [comment]
I think we can wait a few days to integrate that at the same time as we release Circular Dependencies (hopefully this weekend). You might deprecate the old method right away in a 3.1.0 before 4.0.0 but I'm not sure that's worth it.

Also see https://github.com/AliSoftware/Dip/issues/16#issuecomment-159975838 : I couldn't think of a better name than `withArguments:` here, but maybe a little brainstorming is worth it before adopting that to ensure the name we choose is not confusing with a param expecting an array.

**Comment by ilyapuchka** [comment]
I think it's fine - code completion should make it clear that it's different arguments, not an array.

**Comment by AliSoftware** [comment]
True. I was worried about the version with one argument being `Any` it could actually accept an `Array` without complaining; but as you say, code completion will list all the possibilities with one to 6 arguments so this should be clear enough indeed.

**Comment by AliSoftware** [comment]
Dammit, you beat me to editing the wiki page, I conflicted with your edits when wanting to edit it :smile: fast guy :wink:

**Comment by ilyapuchka** [comment]
Last minutes before my wife takes me away to watch TV-show :smile:

**Comment by ilyapuchka** [comment]
Resolved by #21
